### DISCUSSION TITLE
Test for empty slug with dynamic routing so I don't need to have an index.tsx

### DISCUSSION THREAD

**Discussion by kendjr** [vercel/next.js#48272] [Help] [answered]
### Summary

I am using dynamic routing and would like to have a default slug in the event the slug does not match any existing pages. In other words test if the slug is valid, and if not set it to "default".

The file is named "mywebsite.com/wsbm/[slug].tsx".

I have a getStaticPaths function as follows:
```
export const getStaticPaths: GetStaticPaths = async () => {
  return {
    paths: SPONSORS.map((sponsor) => ({
      params: { slug: sponsor.key },
    })),
    fallback: false,
  };
};
```
I then have useRouter as follows:
```
const router = useRouter();
const slug = (router.query.slug) || "default";
   console.log("slug is: " + slug)
   
const currentSponsor = SPONSORS.find((sponsor) => sponsor.key === slug);
  return (

```
When there is no slug "mywebsite.com/wsbm", I receive a 404 error. I have read through the routing documentation, but can't see how to test for a valid slug. Any help would be appreciated!
Ken

**Comment by icyJoseph** [comment]
Hi,

I think you need to use `[[...slug]]`, IIRC:

- https://nextjs.org/docs/routing/dynamic-routes#optional-catch-all-routes

**Reply by kendjr** [reply, marked as answer]
Thanks! 
It was a bit more complex than I thought. I did change it so [[...slug]]. I had to set fallback to true:

```
export const getStaticPaths: GetStaticPaths = async () => {
  const paths = SPONSORS.map((sponsor) => ({
    params: { slug: [sponsor.key] },
  }));
  return {
    paths,
    fallback: true,
  };
};
```
Then set the slug as default, and test for isFallback
```
export default function Index({ allPosts, preview }) {
  const router = useRouter();
  const { isFallback, query } = router;

  const slug = (query.slug as string[]) || ["default"]

  const currentSponsor = SPONSORS.find((sponsor) => slug.includes(sponsor.key)) || SPONSORS.find((sponsor) => sponsor.key === "default");

  if (isFallback) return <div>Loading...</div>
....

```

**Reply by icyJoseph** [reply]
Right, if in the beginning that route is not returned by `getStaticPaths`, you gotta set it fallback true, to allow it _later on_, but that also forces you to handle fallback pages: https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-pages


Good that you figured it all out! Don't forget to mark as answered! Have a good one!
